source/cleo.cpp

```diff
@@ -1,11 +1,59 @@
 #include "stdafx.h"
 #include "cleo.h"
+#include <windows.h>
+#include <shlobj.h>
+
 
 namespace CLEO
 {
     CCleoInstance CleoInstance;
     CCleoInstance& GetInstance() { return CleoInstance; }
 
+    std::string ResolveCleoPath(const char* path)
+    {
+        if (path == nullptr)
+        {
+            return {};
+        }
+
+        if (strlen(path) < 2 || path[1] != ':')
+        {
+            return path;
+        }
+
+        if (path[0] < '0' || path[0] > '2')
+        {
+            return path; // outside supported range
+        }
+
+        std::string resolved(MAX_PATH, '\0');
+
+        if (path[0] == '1')
+        {
+            SHGetFolderPathA(NULL, CSIDL_PERSONAL, NULL, SHGFP_TYPE_CURRENT, resolved.data());
+            resolved.resize(strlen(resolved.data()));
+            resolved += "\\GTA San Andreas User Files";
+            resolved += path + 2; // original path without '1:' prefix
+            return resolved;
+        }
+
+        // game root directory
+        GetModuleFileNameA(NULL, resolved.data(), MAX_PATH);
+        resolved.resize(strlen(resolved.data()));
+        
+        if (path[0] == '2') // cleo directory
+        {
+            resolved += "\\cleo";
+        }
+        else if (path[0] == '3') // cleo modules directory
+        {
+            resolved += "\\cleo\\cleo_modules";
```

Comment:
this whole idea with virtual file system (<num>:\) is interesting, but I think it's worth its own discussion/ implementation. Can we take it out for now to keep modules implementation simple?

Reply:
I had this idea for later, but it turned out I need to build paths in multiple code locations. Single function and custom path format saves simplifies the code.

Reply:
Paths with X:\ prefix, where X is 0,1,2), is just another form of an absolute path. For modules implementation absolute paths are much of a less priority, since modules are meant to be relative to the main file.

We can create a new RFC and discuss this idea further, specifically the proposed syntax and what folders we would like to use (root folder, settings folder, etc).

For the purpose of this MR, this is not needed.

Reply:
This should perhaps be resolved now, as we currently introducing new the third kind of paths: relative to the current script.

Reply:
that is what I meant, relative to the script. There is no other relative paths in module path resolution. They are either absolute (which is bad and not reusable) or relative to the file that imports them.

Reply:
Yes, we are introducing second type of relative paths, as now relative paths are relative to the application's current working directory.